src/pip/_internal/commands/show.py

```diff
@@ -122,47 +148,96 @@ def search_packages_info(query):
         yield package
 
 
-def print_results(distributions, list_files=False, verbose=False):
+def get_package_info(dist, list_files, verbose):
+    """
+    Gather details from an installed distribution.
+    """
+    name = dist.get('name', '')
+    required_by = [
+        pkg.project_name for pkg in pkg_resources.working_set
+        if name in [required.name for required in pkg.requires()]
+    ]
+    info = [
+        ("Name", name),
+        ("Version", dist.get('version', '')),
+        ("Summary", dist.get('summary', '')),
+        ("Home-page", dist.get('home-page', '')),
+        ("Author", dist.get('author', '')),
+        ("Author-email", dist.get('author-email', '')),
+        ("License", dist.get('license', '')),
+        ("Location", dist.get('location', '')),
+        ("Requires", dist.get('requires', [])),
+        ("Required-by", required_by)
+    ]
+
+    if verbose:
+        classifiers = [str(classifier)
+                       for classifier in dist.get('classifiers', [])]
+
+        parser = configparser.ConfigParser()
```

Comment:
Why is this `ConfigParser` stuff here if it wasn't needed in the original implementation?

Reply:
For JSON output, entry points data is needed in a structured format (i.e. a dict of dicts), whereas for text output the entry points data is just dumped out as is. `ConfigParser` does the TOML parsing of the entry points data.